File: TaskFlow/Dtos/Requests/Task/TaskFilterRequest.cs
using TaskFlow.Enums;
using TaskStatus = TaskFlow.Enums.TaskStatus;

namespace TaskFlow.Dtos.Requests.Task;

public class TaskFilterRequest
{
    public TaskStatus? TaskStatus { get; set; }
    public TaskPriority? TaskPriority { get; set; }
    public bool? AssignedToMe { get; set; }
    public DateTime? DueBefore { get; set; }
    public string? Search { get; set; }
}

File: frontend/src/app/core/models/task/task-filter-request.ts
import { TaskPriority } from "../../enums/task-priority-enum";
import { TaskStatus } from "../../enums/task-status-enum";

export interface TaskFilterRequest {
  taskStatus?: TaskStatus;
  taskPriority?: TaskPriority;
  assignedToMe?: boolean;
  dueBefore?: string | null;
  search?: string;
}
